src/v/cluster/members_frontend.cc

```diff
@@ -132,4 +138,79 @@ members_frontend::recommission_node(model::node_id id) {
     }
     co_return res.value().data.error;
 }
+
+ss::future<members_frontend::get_under_replicated_reply>
+members_frontend::get_under_replicated(model::node_id id) {
+    if (id != _self()) {
+        co_return get_under_replicated_reply{
+          .under_replicated = co_await is_node_under_replicated(id),
+          .error = errc::success};
+    }
```

Comment:
is the case where `id != self` important? it seems to me that it also has odd semantics like computing if an under replication partition exists that contains `id` and `self` in the partition's replica set. it looks like this is always called with id == self, too. is this condition used? if not, maybe remove?